src/app/page.tsx

```diff
@@ -1,24 +1,61 @@
-import { CategoriesExpandableList } from "@/components/CategoriesExpandableList";
-import { CategoriesMenu } from "@/components/CategoriesMenu";
-import Header from "@/components/Header";
+"use client";
+
+import { useState } from "react";
+import dynamic from "next/dynamic";
+
+import CategoriesExpandableList from "@/components/CategoriesExpandableList";
 import ListItemCard from "@/components/ListItemCard";
+import { Modal } from "flowbite-react";
+import PreviewItemsAddedSkeleton from "@/components/PreviewItemsAdded/PreviewItemAddedSkeleton";
+
+const PreviewItemAdded = dynamic(
+  () =>
+    import("@/components/PreviewItemsAdded/PreviewItemAdded").then(
+      (module) => module.default
+    ),
+  {
+    loading: () => <PreviewItemsAddedSkeleton />,
+  }
+);
+
+const items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 
 export default function Home() {
-  const items = [1,2,3,4,5,6,7,8,9,10]
+  const [isModalOpen, setIsModalOpen] = useState(false);
```

Comment:
Moved the entire behaviour of the modal to the upper most component that uses it, because having it inside each card will cause a state to be generated for each card, even more, a modal to be generated for each card, which is ineffective.

With this approach, we have 1 modal, with one state controlled inside the page, and managed trough onClick handlers. This can be further isolated into a hook